packages/core/src/lib/cache-layer/cacheLayerLoader.ts

```diff
@@ -0,0 +1,116 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { uniq } from 'lodash';
+import { inject, injectable, interfaces } from 'inversify';
+import { TYPES } from '@vulcan-sql/core/types';
+import {
+  ICacheLayerOptions,
+  cacheProfileName,
+  vulcanCacheSchemaName,
+} from '@vulcan-sql/core/models';
+import { CacheLayerOptions } from '@vulcan-sql/core/options';
+import { APISchema, DataSource } from '@vulcan-sql/core/models';
+import { ConfigurationError } from '../utils/errors';
+
+export interface ICacheLayerLoader {
+  preload(schemas: Array<APISchema>, cacheProfile: string): Promise<void>;
+}
+
+@injectable()
+export class CacheLayerLoader implements ICacheLayerLoader {
+  // Use protected method for unit test
+  protected dataSourceFactory: interfaces.SimpleFactory<DataSource>;
+  private options: ICacheLayerOptions;
+
+  constructor(
+    @inject(TYPES.CacheLayerOptions) options: CacheLayerOptions,
+    @inject(TYPES.Factory_DataSource)
+    dataSourceFactory: interfaces.SimpleFactory<DataSource>
+  ) {
+    this.dataSourceFactory = dataSourceFactory;
+    this.options = options;
+  }
+
+  /**
+   * Preload the cache files to cache data source and created tables
+   * @param schemas The API schemas from artifact to preload
+   */
+  public async preload(schemas: Array<APISchema>): Promise<void> {
+    // prepare cache data source
+    const cacheStorage = this.dataSourceFactory(cacheProfileName);
+    // check if the cache table name is duplicated more than one API schemas
+    this.checkDuplicateCacheTableName(schemas);
+    // traverse each cache table of each schema
+    for (const schema of schemas) {
+      // skip the schema if not set the cache
+      if (!schema.cache) continue;
+      for (const cache of schema.cache) {
+        const { cacheTableName, sql, profile } = cache;
+        const dataSource = this.dataSourceFactory(profile);
+        // directory pattern:[folderPath]/[schema.templateSource]/[profileName]/[cacheTableName]
+        const directory = this.generateDirectory(
+          schema.templateSource,
+          profile,
+          cacheTableName
+        );
+        if (!fs.existsSync(directory!))
+          fs.mkdirSync(directory!, { recursive: true });
+
+        // 1. export to cache files according to each schema set the cache value
+        await dataSource.export({
+          sql,
+          directory,
+          profileName: profile,
+          type: this.options.type!,
+        });
+        // 2. preload the files to cache data source
+        await cacheStorage.import({
+          tableName: cacheTableName,
+          directory,
+          // use the "vulcan.cache" profile to import the cache data
+          profileName: cacheProfileName,
```

Comment:
No, the `profileName` here only accepts the value user set in the profiles, but our `vulcan.cache` is our built-in name actually, so we could pass it here.

Reply:
The `vulcan.cache` is only used for cache layer need, however, we still have other data sources, and not all data sources will use the import method to be the cache layer because the import method could be used to import data from parquet files actually, so user or future work may possibly support import method but not use in the cache layer